tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -245,142 +176,290 @@ def test_many_partitions(self):
-        # is lower (avoid writing excessive data for tests with fewer partitions).
-        # * Then apply a factor of two to make sure we have enough data to drive writes
-        # to disk during consumption, not just enough data to hold it all in the batch
-        # cache.
-        write_bytes_per_topic = min(
-            int((self.redpanda.get_node_memory_mb() * 1024 * 1024) / n_topics),
-            fetch_mb_per_partition * n_partitions) * 2
+            assert len(partitions) == p_per_topic
+            for p in partitions:
+                if p.leader == node_id:
+                    self.logger.info(
+                        f"partition {tn}/{p.id} still on node {node_id}")
+                    any_incomplete = True
 
-        if self.scale.release:
-            # Release tests can be much longer running: 10x the amount of
-            # data we fire through the system
-            write_bytes_per_topic *= 10
+        return not any_incomplete
 
-        msg_size = 128 * 1024
-        msg_count_per_topic = int((write_bytes_per_topic / msg_size))
+    def _node_leadership_balanced(self, topic_names: list[str],
+                                  p_per_topic: int):
+        node_leader_counts = defaultdict(int)
+        any_incomplete = False
+        for tn in topic_names:
+            try:
+                partitions = list(self.rpk.describe_topic(tn, tolerant=True))
+            except RpkException as e:
+                self.logger.warn(f"rpk timeout: {e}")
+                return False
 
-        # Approx time to write or read all messages, for timeouts
-        # Pessimistic bandwidth guess, accounting for the sub-disk bandwidth
-        # that a single-threaded consumer may see
+            if len(partitions) < p_per_topic:
+                self.logger.info(f"describe omits partitions for topic {tn}")
+                any_incomplete = True
+                continue
 
-        expect_bandwidth = 50 * 1024 * 1024
+            assert len(partitions) == p_per_topic
+            for p in partitions:
+                node_leader_counts[p.leader] += 1
 
-        expect_transmit_time = int(write_bytes_per_topic / expect_bandwidth)
-        expect_transmit_time = max(expect_transmit_time, 30)
+        for n, c in node_leader_counts.items():
+            self.logger.info(f"node {n} leaderships: {c}")
 
-        self.logger.info("Entering initial produce")
-        for tn in topic_names:
-            t1 = time.time()
-            producer = FranzGoVerifiableProducer(
-                self.test_context,
-                self.redpanda,
-                tn,
-                msg_size,
-                msg_count_per_topic,
-                custom_node=self.preallocated_nodes)
-            producer.start()
-            producer.wait(timeout_sec=expect_transmit_time)
-            self.free_preallocated_nodes()
-            duration = time.time() - t1
+        if len(node_leader_counts) != len(self.redpanda.nodes):
```

Comment:
An assert that `len(node_leader_counts) <= len(self.redpanda.nodes)` might be useful here.